dandiapi/api/dashboard.py

```diff
@@ -1,22 +1,28 @@
 from allauth.socialaccount.models import SocialAccount
+from django.contrib.auth.mixins import LoginRequiredMixin, UserPassesTestMixin
 from django.contrib.auth.models import User
 from django.core.exceptions import PermissionDenied
 from django.db.models import Exists, OuterRef, Q
 from django.shortcuts import get_object_or_404, render
 from django.views.decorators.http import require_http_methods
 from django.views.generic.base import TemplateView
+from django_filters.views import FilterView
 
+from dandiapi.api.filters import UserStatusFilter
 from dandiapi.api.mail import send_approved_user_message, send_rejected_user_message
 from dandiapi.api.models import Asset, AssetBlob, Upload, UserMetadata, Version
 from dandiapi.api.views.users import social_account_to_dict
 
 
-class DashboardView(TemplateView):
+class DashboardMixin(LoginRequiredMixin, UserPassesTestMixin):
```

Comment:
Use `UserPassesTestMixin` will redirect to the login page if the user doesn't pass the test instead of returning an unauthorized status code. Was that the intention?

Reply:
Yes, at leas that's the behavior of the existing dashboard view (https://github.com/dandi/dandi-api/blob/master/dandiapi/api/dashboard.py#L18-L19 causes the redirect).